src/main/kotlin/at/bitfire/dav4jvm/XmlUtils.kt

```diff
@@ -86,8 +86,14 @@ object XmlUtils {
     /**
      * Same as [readText], but requires a [XmlPullParser.TEXT] value.
      *
+     * **Warning: Using this function is highly not recommended.**
+     *
      * @throws InvalidPropertyException when no text could be read
      */
+    @Deprecated(
+        message = "Use readText instead. This function throws exceptions for unexpected contents.",
+        replaceWith = ReplaceWith("readText(parser)")
+    )
```

Comment:
We should drop it completely.

Reply:
> Same applies to all elements.

Do you mean _all_? Like, make `PropertyFactory.create` return non-null? Or just `Topic`, `GetETag` and `PushSubscribe`?

Reply:
For example `CalendarColor` requires an int color, we can always set to black or something like that but I'm not sure if we want that

Reply:
I'd allow `null` too for the same reason: It may be interesting for an application to know that an empty element was transmitted.

I have thought about returning some kind of for instance `Empty<CalendarColor>` when `CalendarColor` couldn't be parsed with value, but it makes thing unnecessary complex.

The `HrefListProperty` properties should automatically work. The others often already allow null values. I think we should consequently allow null values in all properties.